### PR TITLE
fix(version): enable changing commit message when using amend

### LINKED ISSUE #3936: Using "--amend" and "--message" options will not overwrite the original commit message
## Current Behavior

When using `--amend` and `--message`, the commit will be amended but the message will remain unchanged. The `--message` option is not respected and will be implicitly ignored.

Examples:
 `lerna version minor --amend --message "My commit message"`

 `lerna version publish minor --amend --message "My commit message"`

## Expected Behavior

* The version/publish commands work as usual
* The original commit is amended
* The original commit is updated with the specified message from the flag: `--message`
***

The code below will accomplish everything above, but is not recommended due to pushing remote references twice:
```sh
#!/bin/bash
# 1. Let lerna tag versions, publish npm packages with new versions, then amend commit (using original commit message)
# 2. Push the commit to github. Creates new git tag as well
# 3. Redo the commit message to use a chore commit using some magic
# 4. Push the changes using the new commit message
lerna publish minor --amend --exact --yes
git push --follow-tags --force
version=$(node -p "require('./lerna.json').version")
description=$(cat ./.git/COMMIT_EDITMSG)
git commit --amend --message "[chore] publish packages $version" --message "$description"
git push --force
```

## Steps to Reproduce

* Change a source file in a repository and commit it with message "I changed a file"
* Run `lerna publish minor --amend --exact --message "This is a new commit message" --yes`
* Check repository
  * Git tags should be published correctly ✅ 
  * NPM packages should be published correctly ✅  
  * Original commit amended with versioning changes/only 1 commit exists ✅ 
  * **Commit message should be changed to "This is a new commit message"** ❌ 

This issue may not be prioritized if details are not provided to help us reproduce the issue.

### Failure Logs / Configuration

<details><summary>lerna.json</summary><p>

```json
{
  "npmClient": "yarn",
  "version": "2.934.0",
  "graphType": "all",
  "command": {
    "publish": {
      "registry": "https://npm.pkg.github.com/",
      "graphType": "all",
      "ignoreChanges": [
        "*.md"
      ]
    }
  }
}
```

</p></details>

</p></details>

### Environment

`npx lerna info`:

```
lerna notice cli v4.0.0

 Environment info:
  System:
    OS: macOS 13.6.3
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    Yarn: 1.22.19 - ~/node_modules/.bin/yarn
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
  Utilities:
    Git: 2.39.0 - /usr/local/bin/git
  npmPackages:
    lerna: ^4.0.0 => 4.0.0 
```

### PR DESCRIPTION
**Pull request by amorscher**
allows overriding of the commit message

## Motivation and Context
fixes #3936 

## How Has This Been Tested?
all tests still pass

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (change that has absolutely no effect on users)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by amorscher** [comment]
@JamesHenry I changed the behaviour of --amend in combination with --message so that the message will be overrridden if specified. I think this is a breaking change and therefore not a good solution.

What do you think about using another flag like --amendMessage to support amending and overriding the message in one go?

**Comment by amorscher** [comment]
@fahslaj what do you 💬 nk?

**Comment by JamesHenry** [comment]
Hi @amorscher just following up on the latest plans for this PR please

**Comment by JamesHenry** [comment]
Please feel to reopen this whenever you have chance to work on it, but I'll close it for now for clarity, thanks again

**Comment by amorscher** [comment]
@fahslaj  @JamesHenry Sorry for the delay. I was working on it again but it looks like i am missing rights to reopen the PR

File: libs/commands/version/src/index.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ import { isBreakingChange } from "./lib/is-breaking-change";
 import { makePromptVersion } from "./lib/prompt-version";
 import { remoteBranchExists } from "./lib/remote-branch-exists";
 import { updateLockfileVersion } from "./lib/update-lockfile-version";
+import { measureMemory } from "vm";
```
Comment (ghiscoding) [inline libs/commands/version/src/index.ts:51]: this import is not used anywhere ;)

**Comment by JamesHenry** [comment]
@amorscher No problem, you made some more changes after that comment, but should I infer that this is now ready for review again?

**Comment by amorscher** [comment]
@JamesHenry yes it should be ready for review.

**Comment by JamesHenry** [comment]
Thanks again @amorscher!

**Pull request merged**